core/shared/platform/windows/win_file.c

```diff
@@ -1035,15 +1100,71 @@ os_fallocate(os_file_handle handle, __wasi_filesize_t offset,
 {
     CHECK_VALID_FILE_HANDLE(handle);
 
-    return __WASI_ENOSYS;
+    LARGE_INTEGER current_file_size;
+    GetFileSizeEx(handle->raw.handle, &current_file_size);
```

Comment:
Had better check return value of `GetFileSizeEx`?

Reply:
good spot - will check it